src/game/resource.c

```diff
@@ -74,43 +75,53 @@ static resource_data resource_info[RESOURCE_ALL] = {
     [RESOURCE_WINE]       = { .type = RESOURCE_WINE,       .xml_attr_name = "wine",        .flags = RESOURCE_FLAG_INVENTORY,                      .industry = BUILDING_WINE_WORKSHOP,      .production_per_month = 40,  .default_trade_price = { 215, 160 } },
     [RESOURCE_WEAPONS]    = { .type = RESOURCE_WEAPONS,    .xml_attr_name = "weapons",     .flags = RESOURCE_FLAG_STORABLE,                       .industry = BUILDING_WEAPONS_WORKSHOP,   .production_per_month = 40,  .default_trade_price = { 250, 180 } },
+    [RESOURCE_CONCRETE]   = { .type = RESOURCE_CONCRETE,   .xml_attr_name = "concrete",    .flags = RESOURCE_FLAG_NONE,                           .industry = BUILDING_CONCRETE_MAKER,     .production_per_month = 120 },
+    [RESOURCE_BRICKS]     = { .type = RESOURCE_BRICKS,     .xml_attr_name = "bricks",      .flags = RESOURCE_FLAG_STORABLE,                       .industry = BUILDING_BRICKWORKS,         .production_per_month = 60,  .default_trade_price = { 200, 150 } },
     [RESOURCE_DENARII]    = { .type = RESOURCE_DENARII,    .industry = BUILDING_CITY_MINT, .production_per_month = 200 },
     [RESOURCE_TROOPS]     = { .type = RESOURCE_TROOPS  }
 };
 
-static const resource_chain SUPPLY_CHAIN[] = {
-    { .raw_material = RESOURCE_CLAY,   .good = RESOURCE_POTTERY   },
-    { .raw_material = RESOURCE_TIMBER, .good = RESOURCE_FURNITURE },
-    { .raw_material = RESOURCE_OLIVES, .good = RESOURCE_OIL       },
-    { .raw_material = RESOURCE_VINES,  .good = RESOURCE_WINE      },
-    { .raw_material = RESOURCE_IRON,   .good = RESOURCE_WEAPONS   },
-    { .raw_material = RESOURCE_GOLD,   .good = RESOURCE_DENARII   },
+static const resource_supply_chain SUPPLY_CHAIN[RESOURCE_SUPPLY_CHAIN_MAX_SIZE] = {
+    { .raw_amount = 100, .raw_material = RESOURCE_CLAY,   .good = RESOURCE_POTTERY   },
+    { .raw_amount = 100, .raw_material = RESOURCE_TIMBER, .good = RESOURCE_FURNITURE },
+    { .raw_amount = 100, .raw_material = RESOURCE_OLIVES, .good = RESOURCE_OIL       },
+    { .raw_amount = 100, .raw_material = RESOURCE_VINES,  .good = RESOURCE_WINE      },
+    { .raw_amount = 100, .raw_material = RESOURCE_IRON,   .good = RESOURCE_WEAPONS   },
+    { .raw_amount = 100, .raw_material = RESOURCE_GOLD,   .good = RESOURCE_DENARII   },
+    { .raw_amount =  50, .raw_material = RESOURCE_SAND,   .good = RESOURCE_CONCRETE  },
+    { .raw_amount =  50, .raw_material = RESOURCE_SAND,   .good = RESOURCE_BRICKS    },
+    { .raw_amount =  50, .raw_material = RESOURCE_CLAY,   .good = RESOURCE_BRICKS    },
+    { .raw_amount =  20, .raw_material = RESOURCE_GOLD,   .good = RESOURCE_DENARII   },
 };
 
-#define SUPPLY_CHAIN_SIZE (sizeof(SUPPLY_CHAIN) / sizeof(resource_chain))
-
-static resource_type resource_list[SUPPLY_CHAIN_SIZE + 1];
-
 int resource_is_food(resource_type resource)
 {
     return (resource_info[resource].flags & RESOURCE_FLAG_FOOD) == RESOURCE_FLAG_FOOD;
 }
 
 int resource_is_raw_material(resource_type resource)
```

Comment:
I think this might work better as a hardcoded list.
See example of gold to denarii but also denarii to gold.

Reply:
That is a difficult one.

Gold is an exception and denarii to gold shouldn't be added to the supply chain list.

But besides that exception, raw materials are resources that don't need a previous resource to be obtained, which this code checks.